client/api_client.go

```diff
@@ -160,11 +162,13 @@ func New(chainID string, endpoint string, option Option) (IClient, error) {
 		expireSeconds:    option.ExpireSeconds,
 	}
 
-	// fetch sp endpoints info from chain
-	err = c.refreshStorageProviders(context.Background())
+	if option.Endpoint == "" {
+		// fetch sp endpoints info from chain
```

Comment:
refreshStorageProviders is used to update the following map relationship,
map[uint32]*types.StorageProvider, the complete solution needs to be redesigned. Adding this option alone is useless. If you only need to optimize getObject, you only need to add it to the opt of getObject.

Reply:
Discussed with Leo and would update the Endpoint option to a word that make more sense to users and that make minimal impact to other go-sdk functionalities.